pydriller/repository_mining.py

```diff
@@ -148,6 +150,13 @@ def traverse_commits(self) -> Generator[Commit, None, None]:
 
             logger.info('Analyzing git repository in {}'.format(git_repo.path))
 
+            if self._filepath is not None:
```

Comment:
I wouldn't do it like this, because in this case it can not be used together with other filters. For example, retrieve all the commits of a file between 2 dates, or all the commits of a specific author on the file. What I would do instead is to put a filter at the end of the function `self._is_commit_filtered()`.

Reply:
Absolutely right, Will fix

Reply:
of course, no problems 😄 
thanks!